.github/workflows/python-package.yml

```diff
@@ -675,6 +675,35 @@ jobs:
       if: ${{ contains(needs.*.result, 'failure') }}
       run: exit 1
 
+  doctests:
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    env:
+      python-version: '3.10'
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python ${{ env.python-version }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ env.python-version }}
+        cache: pip
+        cache-dependency-path: |
+          pyproject.toml
+          requirements-dev.txt
+    - name: Setup Virtual Env
+      run: |
+        python -m venv venv
+        echo "$GITHUB_WORKSPACE/venv/bin" >> $GITHUB_PATH
+    - name: Install dependencies
+      run: |
+        pip install --upgrade pip
+        pip install -r requirements-dev.txt
+    - name: Run doctests
+      run: |
+        source activate
+        maturin develop
```

Comment:
Not sure if there's a way to cache the build from a previous test run?

Reply:
Take a look at how we do `rust-cache` in the other workflows: https://github.com/Eventual-Inc/Daft/blob/cd775f58b62bb7e712e29037f4c42bab6ee71a86/.github/workflows/python-package.yml#L45